core/frontend/src/utils/pull_tracker.ts

```diff
@@ -3,10 +3,15 @@ import { Dictionary } from '@/types/common'
 class PullTracker {
     private layer_status: Dictionary<string> = {}
 
+    // We need to assume a size for the missing layers, we are assuming 100mb for now.
+    private guesstimated_size = 100 * 1024 * 1024
+
     private layers: string[] = []
 
     private layer_progress: Dictionary<string> = {}
 
+    private layer_progress_detail: Dictionary<Dictionary<number>> = {}
```

Comment:
Isn't this a `Dictionary<{total: number, current: number}> = {}` ?

Reply:
I didn't remember you could do that. nice.